### ISSUE TITLE
TypeScript error in bridge.d.ts and analytics.d.ts

### ISSUE THREAD

**Issue by cristeahub** [segmentio/analytics-react-native#11]
Hi,

I recently stumbled upon the following error when running `tsc` in my project:

```zsh
node_modules/@segment/analytics-react-native/build/esm/analytics.d.ts:89:26 - error TS1039: Initializers are not allowed in ambient contexts.

89         readonly ready = false;
                            ~~~~~

node_modules/@segment/analytics-react-native/build/esm/bridge.d.ts:2:17 - error TS2503: Cannot find namespace 'NativeModules'.

2 import bridge = NativeModules.RNAnalytics;
                  ~~~~~~~~~~~~~

node_modules/@segment/analytics-react-native/build/esm/bridge.d.ts:4:31 - error TS2503: Cannot find namespace 'NativeModules'.

4 export declare type JsonMap = NativeModules.RNAnalytics.JsonMap;
```

This doesn't cause any errors in the app as far as I can see.

**Comment by fathyb** [comment]
Thanks for the report! The first error is suspicious, what version of TypeScript are you using?

**Comment by cristeahub** [comment]
I'm currenty on `Version 3.0.3`.

**Comment by ajmath** [comment]
I'm getting a similar error from tsc:
```
node_modules/@segment/analytics-react-native/src/bridge.ts:3:17 - error TS2503: Cannot find namespace 'NativeModules'.

3 import bridge = NativeModules.RNAnalytics
                  ~~~~~~~~~~~~~

node_modules/@segment/analytics-react-native/src/bridge.ts:10:23 - error TS2503: Cannot find namespace 'NativeModules'.

10 export type JsonMap = NativeModules.RNAnalytics.JsonMap
                         ~~~~~~~~~~~~~
```

I'm using typescript 3.0.3 with a tsconfig.json that looks like this:
```json
{
  "compilerOptions": {
    "target": "es2017",
    "jsx": "react-native",
    "noEmit": true,
    "moduleResolution": "node",
    "module": "commonjs",
    "strict": true,
    "noUnusedLocals": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "noImplicitReturns": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "resolveJsonModule": true,
    "allowJs": true,
    "checkJs": false,
    "outDir": ".tmp",
    "typeRoots": [
      "node_modules/@types",
      "app/types"
    ],
    "plugins": [
      {
        "name": "tslint-language-service",
        "ignoreDefinitionFiles": true
      }
    ],
    "importHelpers": true
  },
  "include" : [
    "app/**/*",
    "__tests__/**/*"
  ],
  "exclude": [
    "node_modules",
    ".tmp",
    "app/native-base-theme"
  ]
}
```

**Comment by cristeahub** [comment]
I'll have to add I'm on `0.0.1-beta.1` of this lib. I updated my other comment to include my tsconfig.

**Comment by ajmath** [comment]
I was able to resolve this by changing one of my imports from the module from:
`import { JsonMap } from "@segment/analytics-react-native/src/bridge";` to `import { JsonMap } from "@segment/analytics-react-native/build/esm/bridge";`.

I'm not sure if this is possible but maybe you can update the declaration file so that all types can be exported into the default `index.d.ts`?

**Comment by fathyb** [comment]
Thanks @cristeahub! We've [released `0.0.1-beta.2`](https://github.com/segmentio/analytics-react-native/blob/master/CHANGELOG.md#001-beta2-2018-10-24) with new TypeScript tests and the bridge issues fixed.

@ajmath We've also added export to public interfaces so that you can import them using the entry-point:
```ts
import analytics, { Configuration, JsonMap } from '@segment/analytics-react-native'
```
